kernel/src/net/socket/netlink/route/kernel/link.rs

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Handle link-related requests.
+
+use core::num::{NonZero, NonZeroU32};
+
+use aster_bigtcp::iface::InterfaceType;
+
+use crate::{
+    net::{
+        iface::{Iface, IFACES},
+        socket::netlink::route::message::{
+            attr::link::NlLinkAttr, CMessageSegmentHeader, CSegmentType, GetRequestFlags,
+            LinkSegment, LinkSegmentBody, NlSegment, SegmentHeaderCommonFlags,
+        },
+    },
+    prelude::*,
+    util::net::CSocketAddrFamily,
+};
+
+pub fn do_get_link(segment: &NlSegment) -> Result<Vec<NlSegment>> {
+    let NlSegment::Link(request_segment) = segment else {
```

Comment:
Why not match `NlSegment::Link` directly in the `request` method and pass `request_segment` directly?